integration/auth/user_mgt_test.go

```diff
@@ -0,0 +1,421 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package auth contains integration tests for the firebase.google.com/go/auth package.
+package auth
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"google.golang.org/api/iterator"
+
+	"firebase.google.com/go/auth"
+
+	"golang.org/x/net/context"
+)
+
+var testFixtures = struct {
+	uidList            []string
+	sampleUserBlank    *auth.UserRecord
+	sampleUserNil      *auth.UserRecord
+	sampleUserWithData *auth.UserRecord
+}{}
+
+func TestUserManagement(t *testing.T) {
+	t.Run("add some users", createTestUsers)
+	t.Run("created users", createdUsers)
+	t.Run("get user", testGetUser)
+	t.Run("user iterator test", testUserIterator)
+	t.Run("paging iterator test", testPager)
+	t.Run("disable", testDisableUser)
+
+	t.Run("update user", testUpdateUser)
+	t.Run("disable", testDisableUser)
+
+	t.Run("remove PhoneNumber, PhotoURL and DisplaName", testRemovePhonePhotoName)
+
+	t.Run("Remove custom claims", testRemoveCustomClaims)
+	t.Run("add custom claims", testAddCustomClaims)
+
+	t.Run("delete all users", cleanupUsers)
+}
+
+func createTestUsers(t *testing.T) {
+	for i := 0; i < 2; i++ {
+		u, err := client.CreateUser(context.Background(), (&auth.UserToCreate{}).UID(fmt.Sprintf("tempTestUserID-%d", i)))
```

Comment:
Could you extract the UserToCreate initialization to a separate line? Also can we not specify a UID here, and let Firebase Auth generate random UIDs for us?